applications/PoromechanicsApplication/custom_constitutive/isotropic_damage_cohesive_3D_law.cpp

```diff
@@ -276,8 +278,11 @@ void IsotropicDamageCohesive3DLaw::ComputeEquivalentStrain(ConstitutiveLawVariab
     rVariables.OldEquivalentStrain = mOldStateVariable[1] + rVariables.BetaEqStrainShearFactor * mOldStateVariable[0];
 
     // Compute the vector with the derivatives of the equivalent strain wrt to the components of the strain vector
-    rVariables.DerivativeEquivalentStrain[0] = rVariables.BetaEqStrainShearFactor * StrainVector[0] / resShear;
-    rVariables.DerivativeEquivalentStrain[1] = rVariables.BetaEqStrainShearFactor * StrainVector[1] / resShear;
+    rVariables.DerivativeEquivalentStrain = ZeroVector(3);
+    if (resShear > 0.0) {
+        rVariables.DerivativeEquivalentStrain[0] = rVariables.BetaEqStrainShearFactor * StrainVector[0] / resShear;
+        rVariables.DerivativeEquivalentStrain[1] = rVariables.BetaEqStrainShearFactor * StrainVector[1] / resShear;
+    }
```

Comment:
I'm not familiar with the physics here, but what happens if `resShear == 0.0`? Is the model still valid? If not, this just silently swallows the error which is going to be a nightmare to debug when you only see results that don't make sense even though the program didn't crash / throw an exception.

Reply:
What this PR is doing is exactly solving the error when the resShear is equal to zero. For this case, the components of the vector DerivativeEquivalentStrain will keep with their initial value, as zero (which is not a problem).

Reply:
> For this case, the components of the vector DerivativeEquivalentStrain will keep with their initial value, as zero (which is not a problem).

allright